lib/pbench/server/api/resources/datasets_list.py

```diff
@@ -348,20 +354,77 @@ def keyspace(self, query: Query) -> JSONOBJECT:
                     self.accumulate(aggregate, m.key, m.value)
         return aggregate
 
-    def datasets(self, request: Request, json: JSONOBJECT, query: Query) -> JSONOBJECT:
+    def datasets(
+        self, request: Request, aliases: dict[str, Any], json: JSONOBJECT, query: Query
+    ) -> JSONOBJECT:
         """Gather and paginate the selected datasets
 
         Run the query we've compiled, with pagination limits applied; collect
         results into a list of JSON objects including selected metadata keys.
 
         Args:
             request: The HTTP Request object
+            aliases: Map of join column aliases for each Metadata namespace
             json: The JSON query parameters
             query: The basic filtered SQLAlchemy query object
 
         Returns:
             The paginated dataset listing
         """
+
+        # Process a possible list of sort terms. By default, we sort by the
+        # dataset resource_id.
+        sorters = []
+        for sort in json.get("sort", ["dataset.resource_id"]):
+            if ":" not in sort:
+                k = sort
+                order = asc
+            else:
+                try:
+                    k, o = sort.split(":", maxsplit=1)
+                except ValueError:
+                    raise APIAbort(
+                        HTTPStatus.BAD_REQUEST,
+                        f"sort {sort!r} must have the form 'k[:o]'",
+                    )
```

Comment:
Given that line 379 ensures that there is at least one colon in `sort`, how would you ever get a `ValueError` from line 384?  (I don't think the `try` block at line 383 is needed....)

Reply:
Right: when I made the sort direction (and `:`) optional, I should have changed this.